### PR TITLE
QC addresses issues raised in #28

### LINKED ISSUE #30: Address comments on #28
Please address my comments on #28.

### PR DESCRIPTION
**Pull request by mqchau**
Closes #30

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
     end
 
     assert_redirected_to Image.last
+    get image_url(Image.last)
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: Please don't do this. Controller tests should represent only one request.

File: app/views/layouts/_flash_message.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<% flash.each do |key, value| %>
```
Comment (bboe) [inline app/views/layouts/_flash_message.html.erb]: ⭐

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -91,15 +92,13 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
-    get image_url(image)
-    assert_response :not_found
   end
 
   test 'should not delete image with invalid id' do
-    assert_difference('Image.count', 0) do
-      delete image_url(-1)
+    assert_no_difference('Image.count') do
+      assert_raises ActiveRecord::RecordNotFound do
+        delete image_url(-1)
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: ⭐

File: test/flows/image_flow_test.rb [hunk 4]
```diff
@@ -2,12 +2,18 @@
 
 class ImageFlowTest < FlowTestCase
   test 'add new image then delete' do
+    sample_img_link = 'http://i.imgur.com/PVrX7wZ.jpg'
```
Comment (bboe) [inline test/flows/image_flow_test.rb]: Great updates.

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -26,6 +26,7 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
     assert_redirected_to Image.last
     get image_url(Image.last)
     assert_equal 'Image was successfully created.', flash[:success]
+    assert_select '.image__tag', 'testtag1'
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: These updates should be unnecessary due to not making a second request in the controller.

**Comment by bboe** [comment]
@mqchau I added my feedback.

**Comment by mqchau** [comment]
I think I have addressed all your concerns, Bryce.

File: test/controllers/images_controller_test.rb [hunk 6]
```diff
@@ -24,7 +24,6 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
     end
 
     assert_redirected_to Image.last
-    assert_equal 'Image was successfully created.', flash[:success]
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: Why were the flash assertions removed? Flash messages (cookies) are side effects that should be tested for.

**Comment by bboe** [comment]
One last comment. After fixing please atomic-ify your commits on this pull request.

**Comment by mqchau** [comment]
I fixed it.

**Comment by bboe** [comment]
Nearly perfect.

>   - look up Image in db using find and catch error ActiveRecord NotFound in controller test

Is that still something that's being done in the commit?

Also, can you provide a more descriptive summary line than closes #28? Pretend in the future that github goes away and we have no way to know what #28 is.

**Comment by bboe** [comment]
There should also be a comment in the comment as to why selenium webdriver needed to be bumped.

**Comment by mqchau** [comment]
Yes this commit has the part of "look up Image in db using find and catch error ActiveRecord NotFound in controller test" to undo the last commit change.

**Comment by bboe** [comment]
👍 ready to merge

**Pull request merged**